src/main/java/io/curity/authenticator/netid/injectors/NetIdAccessServerSoapFactory.java

```diff
@@ -146,4 +164,45 @@ private String buildEndpointAddress(String scheme)
             throw new IllegalArgumentException("Could not build URL to Net iD Access server: " + e.getMessage(), e);
         }
     }
+
+    private static SSLContext getCustomSSLContext(
+            Optional<ServerTrustCryptoStore> trustCryptoStore,
+            Optional<ClientKeyCryptoStore> keyCryptoStore)
+
+    {
+        @Nullable TrustManager[] trustManagers = null;
+        @Nullable KeyManager[] keyManagers = null;
+
+        try
+        {
+            if (trustCryptoStore.isPresent()) {
+                _logger.debug("Applying ssl server-truststore from configuration.");
+
+                var trustStore = trustCryptoStore.get().getAsKeyStore();
+
+                var trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                trustManagerFactory.init(trustStore);
```

Comment:
This might be a bit heavy, and we do it on every request. Perhaps we should move this to a `ManagedObject` instead?